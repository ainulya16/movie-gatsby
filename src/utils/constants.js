export const theme = {
  color: {
    w: '#FFFFFF',
    k: '#000000',
    p1: '#D31145',
    p2: '#596C80',
    p3: '#554344',
    p4: '#363E3F',
    s1: '#E86487',
    s2: '#FBBF56',
    s3: '#97CB5D',
    s4: '#46D3BD',
    s5: '#22A8DA',
    s6: '#9962D2',
    t2: '#B8123E',
    b2: '#FAF9F3',
    b3: '#F0EEE1',
    b5: '#B3B1A8',
    u2: '#ce3c3a',
    u3: '#405B70',
    u4: '#769FAB',
    u5: '#c1272d',
    w9: '335px',
    b9: '#EEF0F2',
    whatsapp: '#25d366',
  },
};

export default null;
