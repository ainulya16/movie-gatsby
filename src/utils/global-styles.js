import { createGlobalStyle } from 'styled-components';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import 'antd/dist/antd.css';
// import '../assets/css/slick.min.css';

const GlobalStyle = createGlobalStyle`
  body {
    font-family: 'AIAMedium';
  }
  @media (max-width: 575px) {
    .wrap_center .ant-form-item-label {
      text-align: center !important;
    }
  }
`;

export default GlobalStyle;
