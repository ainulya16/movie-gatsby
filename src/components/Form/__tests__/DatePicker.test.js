import React from 'react';
import { shallow } from 'enzyme';
import { Field } from 'redux-form/immutable';

// import jest-styled-components for snapshot styled components
import 'jest-styled-components';
import renderer from 'react-test-renderer';

// import components to be tested
import InputComponent, {
  StyledFormItem,
  StyledInput,
  InputTest,
} from '../DatePicker';
import { theme } from '../../../utils/constants';

describe('Input Component', () => {
  it('should have redux-form Field"', () => {
    const component = shallow(<InputComponent name="test" />);

    expect(component.find(Field).exists()).toBe(true);
  });

  it('should render StyledFormItem styles correctly', () => {
    const tree = renderer.create(<StyledFormItem theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render StyledInput styles correctly', () => {
    const tree = renderer.create(<StyledInput theme={theme} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  describe('Input', () => {
    it('must be exists', () => {
      const mockProps = {
        input: {
          name: 'test-name',
          value: 'test value',
        },
        meta: {
          touched: false,
          invalid: false,
          error: 'MOCK ERROR',
        },
      };
      const wrapper = shallow(<InputTest {...mockProps} />);
      expect(wrapper.exists()).toBe(true);
    });
    describe('When hasError is true', () => {
      const mockProps = {
        input: {
          name: 'test-name',
          value: 'test value',
        },
        meta: {
          touched: true,
          invalid: true,
          error: 'MOCK ERROR',
        },
      };
      const wrapper = shallow(<InputTest {...mockProps} />);
      it('should be error in StyledFormItem', () => {
        expect(wrapper.find(<StyledFormItem help label />));
      });
      it('should have a correct validate status', () => {
        expect(wrapper.find(<StyledFormItem validateStatus />));
      });
    });
    describe('When hasError is false', () => {
      const mockProps = {
        input: {
          name: 'test-name',
          value: 'test value',
        },
        meta: {
          touched: true,
          invalid: false,
          error: 'MOCK ERROR',
        },
      };
      const wrapper = shallow(<InputTest {...mockProps} />);
      it('should have no error in StyledFormItem', () => {
        expect(wrapper.find(<StyledFormItem help />));
      });
      it('should have a correct validate status', () => {
        expect(wrapper.find(<StyledFormItem validateStatus />));
      });
    });
    describe('When hasError and hasFeedback are true', () => {
      const mockProps = {
        input: {
          name: 'test-name',
          value: 'test value',
        },
        meta: {
          touched: true,
          invalid: true,
          error: 'MOCK ERROR',
        },
        hasFeedback: true,
      };
      const wrapper = shallow(<InputTest {...mockProps} />);
      it('should have error in StyledFormItem', () => {
        expect(wrapper.find(<StyledFormItem hasFeedback />));
      });
    });
  });
});
