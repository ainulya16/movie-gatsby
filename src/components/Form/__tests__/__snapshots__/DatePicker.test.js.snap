// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render StyledFormItem styles correctly 1`] = `
.c0 > .ant-form-item-control-wrapper > .ant-form-item-control.has-error > span > input {
  border: 1 px solid #FF5500;
}

.c0 .ant-calendar-picker-input {
  width: 335px;
  height: 40px !important;
  font: normal 14px "AIABody",Arial,"Helvetica Neue",Helvetica,sans-serif;
  color: #797979 !important;
  background-color: #EEF0F2;
}

<div
  className="ant-row ant-form-item c0"
>
  <div
    className="ant-col ant-form-item-control-wrapper"
  >
    <div
      className="ant-form-item-control"
    >
      <span
        className="ant-form-item-children"
      />
    </div>
  </div>
</div>
`;

exports[`Input Component should render StyledInput styles correctly 1`] = `
.c1 .ant-calendar-picker-icon {
  display: none;
}

.c1 .ant-calendar-picker-clear {
  display: none;
}

.c1 input {
  font: normal 16px 'AIAMedium',Arial,'Helvetica Neue',Helvetica,sans-serif;
  border: unset;
  background-color: #eef0f1;
  height: 40px;
  padding-left: 16px;
  padding-right: 16px;
  border-radius: 17px;
  background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA0NSA0NSI+PHBvbHlnb24gZmlsbD0iIzU5NkM4MCIgcG9pbnRzPSIxMi42IDYuNCAxMi42IDE0LjkgMTUuNCAxNC45IDE1LjQgMTIuMSAyOS42IDEyLjEgMjkuNiA5LjIgMTUuNCA5LjIgMTUuNCA2LjQgIi8+PHJlY3QgeD0iMTIuNiIgeT0iMjMuNCIgZmlsbD0iIzU5NkM4MCIgd2lkdGg9IjIuOCIgaGVpZ2h0PSIyLjgiLz48cmVjdCB4PSIxOC4zIiB5PSIyMy40IiBmaWxsPSIjNTk2QzgwIiB3aWR0aD0iMi44IiBoZWlnaHQ9IjIuOCIvPjxyZWN0IHg9IjIzLjkiIHk9IjIzLjQiIGZpbGw9IiM1OTZDODAiIHdpZHRoPSIyLjgiIGhlaWdodD0iMi44Ii8+PHJlY3QgeD0iMjkuNiIgeT0iMjMuNCIgZmlsbD0iIzU5NkM4MCIgd2lkdGg9IjIuOCIgaGVpZ2h0PSIyLjgiLz48cmVjdCB4PSIxMi42IiB5PSIyOS4xIiBmaWxsPSIjNTk2QzgwIiB3aWR0aD0iMi44IiBoZWlnaHQ9IjIuOCIvPjxyZWN0IHg9IjE4LjMiIHk9IjI5LjEiIGZpbGw9IiM1OTZDODAiIHdpZHRoPSIyLjgiIGhlaWdodD0iMi44Ii8+PHJlY3QgeD0iMjMuOSIgeT0iMjkuMSIgZmlsbD0iIzU5NkM4MCIgd2lkdGg9IjIuOCIgaGVpZ2h0PSIyLjgiLz48cmVjdCB4PSIyOS42IiB5PSIyOS4xIiBmaWxsPSIjNTk2QzgwIiB3aWR0aD0iMi44IiBoZWlnaHQ9IjIuOCIvPjxwYXRoIGZpbGw9IiM1OTZDODAiIGQ9Ik0zOC4xIDE5LjJ2LTIuOFY5LjJoLTQuM1Y2LjRoLTIuOHY4LjVoMi44di0yLjhoMS40djQuM0g5Ljh2LTQuM2gxLjRWOS4ySDYuOXY3LjEgMi44IDE5LjloMzEuMlYxOS4yek0zNS4zIDM2LjJIOS44VjE5LjJoMjUuNVYzNi4yeiIvPjwvc3ZnPg==');
  background-repeat: no-repeat;
  background-position: calc(100% - 12px) 50%;
  background-size: 20px;
}

.c1 input:focus {
  border: 1px solid #b3b1a8;
}

.c1 .ant-calendar-selected-day div {
  background-color: red !important;
}

.c1 .ant-calendar {
  font: normal 16px 'AIAMedium',Arial,'Helvetica Neue',Helvetica,sans-serif;
}

.c1 .ant-calendar-date-input-wrap {
  font: normal 16px 'AIAMedium',Arial,'Helvetica Neue',Helvetica,sans-serif;
}

.c0.c0 .ant-input {
  padding: 8px 16px;
  border-radius: 15px;
  width: 335px;
  height: 40px !important;
  font: normal 14px "AIABody",Arial,"Helvetica Neue",Helvetica,sans-serif;
  line-height: 17px !important;
  color: #797979 !important;
  background-color: #EEF0F2;
  font-weight: 400;
  border: none !important;
}

.c0.c0 .ant-input:focus {
  color: #797979 !important;
  border: none !important;
  outline: none;
  -webkit-box-shadow: none;
  box-shadow: none;
}

<span
  className="c0 c1 ant-calendar-picker"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={Object {}}
>
  <div
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <input
      className="ant-calendar-picker-input ant-input"
      placeholder="Select date"
      readOnly={true}
      value=""
    />
    <i
      aria-label="icon: calendar"
      className="anticon anticon-calendar ant-calendar-picker-icon"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="calendar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </i>
  </div>
</span>
`;
